nix/build-vars.nix

```diff
@@ -0,0 +1,10 @@
+# DO NOT EDIT, this file is auto-generated by
+# generate-build-vars.sh for purposes of getting around
+# the dependency our build has on the .git folder, which is
+# not copied to the nix store when building. See the README for
+# more details.
+{
+  imageTag = "main-5a6d09f";
+  gitRevision = "5a6d09f83";
+  gitBranch = "main";
```

Comment:
in general there probably isn't much need to commit changes to this file, so I decided to have it track `main`. The `gitRevision` is only used on dirty git trees, and there's a shell hook that will automatically update these values for your local branch when you run `nix develop`

Reply:
What is the reasoning to have it in git?

I suspect to still be able to build the derivation when the flake is imported. I would prefer to have a `unknown` revision instead. As otherwise it is just implying the wrong thing.